element_facemap/facial_behavior_estimation.py

```diff
@@ -0,0 +1,361 @@
+        This user-provided function specifies where Facemap output files
+        will be stored.
+    """
+    if hasattr(_linking_module, 'get_facemap_processed_data_dir'):
+        return _linking_module.get_facemap_processed_data_dir()
+    else:
+        return get_facemap_root_data_dir()[0]
+
+
+def get_facemap_video_files(video_key: dict) -> str:
+    """
+    Retrieve the list of video files (e.g. *.avi) associated with a given video recording
+    :param video_key: key of a video recording
+    :return: list of Video files' full file-paths
+    """
+    return _linking_module.get_facemap_video_files(video_key)
+
+
+# ----------------------------- Table declarations ----------------------
+
+
+@schema
+class VideoRecording(dj.Manual):
+    definition = """
+    -> Session
+    -> Device
+    recording_id                : int
+    """
+
+    # One VideoRecording can be saved in multiple files
+    class File(dj.Part):
+        definition = """
+        -> master
+        file_path: varchar(255)     # filepath of video, relative to root data directory
+        """
+
+
+@schema
+class RecordingInfo(dj.Imported):
+    definition = """
+    -> VideoRecording
+    ---
+    px_height                 : smallint  # height in pixels
+    px_width                  : smallint  # width in pixels
+    nframes                   : smallint  # number of frames 
+    fps                       : float     # (Hz) frames per second
+    recording_duration = NULL : float     # video duration in seconds
+    recording_time = NULL     : datetime  # Time at the beginning of the recording with respect to the session
+    """
+
+    @property
+    def key_source(self):
+        return VideoRecording & VideoRecording.File
+
+    def make(self, key):
+        file_paths = (VideoRecording & key).fetch('file_path')
+
+        nframes = 0
+        px_height, px_width, fps = None, None, None
+
+        for file_path in file_paths:
+            file_path = (find_full_path(get_facemap_root_data_dir(), file_path)).as_posix()
+            # Open the video
+            with cv2.VideoCapture(file_path) as cap:
+                info = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT)), int(cap.get(cv2.CAP_PROP_FRAME_WIDTH)), int(cap.get(cv2.CAP_PROP_FPS))
+                if px_height is not None:
+                    assert (px_height, px_width, fps) == info
+                px_height, px_width, fps = info
+                nframes += int(cap.get(cv2.CAP_PROP_FRAME_COUNT))
+
+        self.insert1({
+            **key,
+            'px_height': px_height,
+            'px_width': px_width,
+            'nframes': nframes,
+            'fps': fps,  # Assuming that this keyword is set correctly but it's usually not.
+            'duration': nframes / fps,                     # TODO Something to think about.
+        })
+    
+
+@schema
+class FacemapParamSet(dj.Manual):
+    definition = """
+    # Parameters used to run the Facemap on each Recording
+    -> VideoRecording
```

Comment:
Under your current design, would someone need to re-enter their params for each video? I think this table might work better as a LookUp

For deeplabcut, we separate a [paramset table](https://github.com/datajoint/element-deeplabcut/blob/main/element_deeplabcut/train.py#L114), so that the [tasks table](https://github.com/datajoint/element-deeplabcut/blob/main/element_deeplabcut/train.py#L178) inherits pairings of paramsets and videos.

Reply:
Each video will have its own parameter set. A parameter set includes parameters (e.g. pixel indices) that define one or several regions (usually box(es)) in a given video.